eos_data_distribution/DirTools/__init__.py

```diff
@@ -20,6 +20,11 @@ def __init__(self, dir, flags=Gio.FileMonitorFlags.NONE, userdata=None):
                 'Asked to monitor something that is not a dir: %s' % (dir, ))
         self.monitorAll(f)
 
+    # XXX: this is needed because older version of Gio.Monitor implement
+    # different signals.
+    def filterSignals(self, a):
+        return [e for e in a if e in EventToSignal.keys()]
```

Comment:
Something like the following might be clearer, although you should check the types:
```
return set(a).intersection(EventToSignal.keys())
```